nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java

```diff
@@ -84,6 +85,12 @@ private TableMetadata loadTableMetadata(String metadataLocation, Reference refer
         NessieUtil.tableMetadataFromIcebergTable(io(), table, metadataLocation);
     Map<String, String> newProperties = Maps.newHashMap(deserialized.properties());
     newProperties.put(NESSIE_COMMIT_ID_PROPERTY, reference.getHash());
+    // To prevent accidental deletion of files that are still referenced by other branches/tags,
+    // setting GC_ENABLED to false. So that all Iceberg's gc operations like expire_snapshots,
+    // remove_orphan_files, drop_table with purge will fail with an error.
+    // Nessie CLI will provide a reference aware GC functionality for the expired/unreferenced
+    // files.
+    newProperties.put(TableProperties.GC_ENABLED, "false");
```

Comment:
This way we disable the GC'ing of the tables completely. Are there cases where we still want Iceberg to do the garbage collection?

Reply:
That makes sense, thanks!